ipn/ipnlocal/local.go

```diff
@@ -2325,10 +2326,14 @@ func dnsConfigForNetmap(nm *netmap.NetworkMap, prefs *ipn.Prefs, logf logger.Log
 			dcfg.Routes[dom] = nil // resolve internally with dcfg.Hosts
 		}
 	}
+	if cloud == cloudenv.GCP {
```

Comment:
Maybe a comment explaining the intent? Without knowing the background, I'd expect the preexisting VM DNS config to cover this, so why do we need to care? I think we want this half so that "override local DNS" keeps working, and half to work around broken base configs, right?